### PR TITLE
Add checks execution domain logic

### PR DESCRIPTION
**Pull request by fabriziosestito**
This PR is the first iteration of the runner checks execution.
It adds the domain logic in the cluster aggregate and the commands and events needed to perform an execution process.
It also _temporarily_  fixes the old FE so we can have some user feedback (review, yay).

ATM we are leveraging the `MockRunner` in the development environment and ignoring the host connection settings.
More to come!

### CODE REVIEW

File: lib/trento/application/integration/checks/adapter/mock_runner.ex [hunk 1]
```diff
@@ -68,21 +74,58 @@ defmodule Trento.Integration.Checks.MockRunner do
     GenServer.cast(__MODULE__, {:set_expected_results, expected_results})
   end
 
-  defp send_execution_started_event(execution_id) do
-    # TODO: post event to callback
+  defp send_execution_started_event(execution_id, cluster_id) do
+    HTTPoison.post!(
+      "http://localhost:4000/api/runner/callback",
```
Comment (arbulu89) [inline lib/trento/application/integration/checks/adapter/mock_runner.ex:79]: Let's put a `TODO` maybe here. This hardcoded address looks something we need to change

File: lib/trento/application/integration/checks/adapter/mock_runner.ex [hunk 2]
```diff
@@ -68,21 +74,58 @@ defmodule Trento.Integration.Checks.MockRunner do
...
+                    }
+                  end)
+              }
+            end)
+        }
+      }),
+      [{"Content-type", "application/json"}]
     )
+
+    # TODO: post event to callback
```
Comment (fabriziosestito) [inline lib/trento/application/integration/checks/adapter/mock_runner.ex:125]: remove TODO

File: lib/trento/application/integration/checks/adapter/mock_runner.ex [hunk 3]
```diff
@@ -68,21 +74,58 @@ defmodule Trento.Integration.Checks.MockRunner do
...
   end
 
-  defp send_execution_started_event(execution_id) do
-    # TODO: post event to callback
+  defp send_execution_started_event(execution_id, cluster_id) do
+    HTTPoison.post!(
+      "http://localhost:4000/api/runner/callback",
+      Jason.encode!(%{
+        "event" => "execution_started",
+        "cluster_id" => cluster_id,
```
Comment (arbulu89) [inline lib/trento/application/integration/checks/adapter/mock_runner.ex:82]: We need to implement this in the runner part. I have already forgotten hehe

File: lib/trento_web/controllers/cluster_controller.ex [hunk 4]
```diff
@@ -6,6 +6,8 @@ defmodule TrentoWeb.ClusterController do
     Tags
   }
 
+  alias Trento.Integration.Checks
```
Comment (arbulu89) [inline lib/trento_web/controllers/cluster_controller.ex:9]: Should maybe this go in a new `ChecksController` file?

**Review by arbulu89 — COMMENTED** [review]
Amazing, amazing job @fabriziosestito 
I have noted down some discussion points, but the change looks super promising.

PD: This looks a crutial part of the project, and I think a mob review would be ideal

File: lib/trento/application/integration/checks/adapter/mock_runner.ex (hunk 1 above)
Comment (fabriziosestito) [reply lib/trento/application/integration/checks/adapter/mock_runner.ex:79]: well, it's a mock we can make it dynamic ofc

File: assets/js/components/ChecksResults.jsx [hunk 5]
```diff
@@ -55,6 +31,12 @@ const getHostname =
     }, '');
   };
 
+const getCatalogByProvider = (catalogProvider) => (state) => {
```
[the remark this answers is not in the document]
Comment (fabriziosestito) [reply assets/js/components/ChecksResults.jsx:34]: alright!

File: lib/trento_web/controllers/cluster_controller.ex (hunk 4 above)
Comment (fabriziosestito) [reply lib/trento_web/controllers/cluster_controller.ex:9]: #259

**Review by nelsonkopliku — APPROVED** [review]
Crazy journey :smile: 

Great iteration!

**Review by dottorblaster — APPROVED** [review]
YES

Wow you did actually fry your brain for this. I'm impressed! :laughing:

File: lib/trento/application/read_models/check_result_read_model.ex [hunk 6]
```diff
@@ -17,9 +15,8 @@ defmodule Trento.CheckResultReadModel do
     field :cluster_id, Ecto.UUID, primary_key: true
     field :host_id, Ecto.UUID, primary_key: true
     field :check_id, :string, primary_key: true
-    field :result, Ecto.Enum, values: [:passing, :warning, :critical, :running]
+    field :result, Ecto.Enum, values: [:passing, :warning, :critical, :unknown]
```
Comment (arbulu89) [inline lib/trento/application/read_models/check_result_read_model.ex:18]: Other small comment. From the runner, we pass the `skipped` result as well to show which were the skipped checks.

Comment (fabriziosestito) [reply lib/trento/application/read_models/check_result_read_model.ex:18]: noted, will come in follow up PRs

**Pull request merged**